es-notebooks/src/test/kotlin/com/amazon/opendistroforelasticsearch/notebooks/PluginRestTestCase.kt

```diff
@@ -0,0 +1,232 @@
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ *
+ */
+
+package com.amazon.opendistroforelasticsearch.notebooks
+
+import com.google.gson.JsonObject
+import org.apache.http.Header
+import org.apache.http.HttpHost
+import org.apache.http.auth.AuthScope
+import org.apache.http.auth.UsernamePasswordCredentials
+import org.apache.http.client.CredentialsProvider
+import org.apache.http.client.config.RequestConfig
+import org.apache.http.conn.ssl.NoopHostnameVerifier
+import org.apache.http.impl.client.BasicCredentialsProvider
+import org.apache.http.impl.nio.client.HttpAsyncClientBuilder
+import org.apache.http.message.BasicHeader
+import org.apache.http.ssl.SSLContextBuilder
+import org.elasticsearch.client.Request
+import org.elasticsearch.client.RequestOptions
+import org.elasticsearch.client.Response
+import org.elasticsearch.client.ResponseException
+import org.elasticsearch.client.RestClient
+import org.elasticsearch.client.RestClientBuilder
+import org.elasticsearch.common.settings.Settings
+import org.elasticsearch.common.unit.TimeValue
+import org.elasticsearch.common.util.concurrent.ThreadContext
+import org.elasticsearch.common.xcontent.DeprecationHandler
+import org.elasticsearch.common.xcontent.NamedXContentRegistry
+import org.elasticsearch.common.xcontent.XContentType
+import org.elasticsearch.test.rest.ESRestTestCase
+import org.junit.After
+import org.junit.Before
+import java.io.BufferedReader
+import java.io.IOException
+import java.io.InputStreamReader
+import java.nio.charset.StandardCharsets
+import java.security.cert.X509Certificate
+
+abstract class PluginRestTestCase : ESRestTestCase() {
+
+    private fun isHttps(): Boolean {
+        return System.getProperty("https", "false")!!.toBoolean()
+    }
+
+    override fun getProtocol(): String {
+        return if (isHttps()) {
+            "https"
+        } else {
+            "http"
+        }
+    }
+
+    override fun preserveIndicesUponCompletion(): Boolean {
+        return true
+    }
+
+    @Throws(IOException::class)
+    @After
+    open fun wipeAllODFEIndices() {
+        val response = client().performRequest(Request("GET", "/_cat/indices?format=json&expand_wildcards=all"))
+        val xContentType = XContentType.fromMediaTypeOrFormat(response.entity.contentType.value)
+        xContentType.xContent().createParser(
+            NamedXContentRegistry.EMPTY, DeprecationHandler.THROW_UNSUPPORTED_OPERATION,
+            response.entity.content
+        ).use { parser ->
+            for (index in parser.list()) {
+                val jsonObject: Map<*, *> = index as java.util.HashMap<*, *>
+                val indexName: String = jsonObject["index"] as String
+                // .opendistro_security isn't allowed to delete from cluster
+                if (".opendistro_security" != indexName) {
```

Comment:
np: can we format it as `indexName != ".opendistro_security"`?

Reply:
Will do. BTW this test setup is copied from AK's reporting test setup, so if you have reviewed that you don't have to review the setup files again